0001-typedb/04-tutorials/01-data-migration.md

```diff
@@ -0,0 +1,868 @@
+
+
+<!--  
+This syntax doesn't work for a single row table without titles:
+| **CSV** | [calls.csv](https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/calls.csv) | [companies.csv](https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/companies.csv) | [contracts.csv](https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/contracts.csv) | [people.csv](https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/people.csv) |
+--->
+
+#### CSV
+
+<table>
+  <tr>
+   <td> CSV
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/calls.csv">calls.csv</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/companies.csv">companies.csv</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/contracts.csv">contracts.csv</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/people.csv">people.csv</a>
+   </td>
+  </tr>
+</table>
+
+#### JSON
+
+<table>
+  <tr>
+   <td>JSON
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/calls.json">calls.json</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/companies.json">companies.json</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/contracts.json">contracts.json</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/people.json">people.json</a>
+   </td>
+  </tr>
+</table>
+
+#### XML
+
+<table>
+  <tr>
+   <td> XML
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/calls.xml">calls.xml</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/companies.xml">companies.xml</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/contracts.xml">contracts.xml</a>
+   </td>
+   <td><a href="https://github.com/vaticle/typedb-examples/blob/master/telecom/phone_calls/data/people.xml">people.xml</a>
+   </td>
+  </tr>
+</table>
+
+## Application
+
+In this example we will conduct migration by building an application to read through dataset files above and insert 
+data into the TypeDB database created. 
+
+The Phone Calls example application showcases:
+
+* importing data into a TypeDB database from:
+    * [CSV](#csv)
+    * [JSON](#json)
+    * [XML](#xml)
+* performing TypeQL queries to the database with imported data.
+
+### Prerequirements
+
+Use TypeDB server (v.**2.11.1+**), Java version **11+** and any IDE, for example, IntelliJ IDEA.
```

Comment:
You could showcase a newer version, will that be updated to the latest at some point? How will we maintain compatibility & accuracy here?

Reply:
We can't do it here, as there are Bazel, node.js, and npm dependencies for the typedb-client of this version. I have used the phone_calls example that was built that way. Maybe sometime later someone can update the example or we will write a different example for this tutroial. But it's out of scope for the current task.